cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java

```diff
@@ -139,6 +143,38 @@ private void initActionFailedExceptionConversion() {
                 .thenAnswer(invocation -> new Retry.ActionFailedException(new CloudConnectorException(invocation.getArgument(0, Exception.class))));
     }
 
+    private static Object[][] testDiskEncryptionSetResourceGroupNameExtractionDataProvider() {
+        return new Object[][]{
+                //testCaseName     diskEncryptionSetResourceId     expectedResourceGroupName}
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaResourceGroup/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaResourceGroup"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaNumResourceGroup0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumResourceGroup0910"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaNumHyphenResourceGroup---0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumHyphenResourceGroup---0910"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaNumHyphenPeriodResourceGroup---....0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaNumHyphenPeriodResourceGroup---....0910"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyAlphaParenthesesPeriodResourceGroup---....)()(0910"},
+                {"testDiskEncryptionSetResourceGroupName", "/subscriptions/dummySubscriptionId/resourceGroups/" +
+                        "dummyUnderscoreResourceGroup---___....)()(0910/providers/Microsoft.Compute/diskEncryptionSets/dummyDes",
+                        "dummyUnderscoreResourceGroup---___....)()(0910"}
+        };
+    }
+
+    @ParameterizedTest(name = "{0}")
```

Comment:
the name should show what you are testing with some description. This is just a suggestion